src/main/scala/gitbucket/core/service/ActivityService.scala

```diff
@@ -97,478 +98,6 @@ trait ActivityService {
-        None,
-        currentDate,
-        UUID.randomUUID().toString
-      )
-    )
-  }
-
-  def recordDeleteBranchActivity(
-    userName: String,
-    repositoryName: String,
-    activityUserName: String,
-    branchName: String
-  ): Unit = {
-    writeLog(
-      Activity(
-        userName,
-        repositoryName,
-        activityUserName,
-        "delete_branch",
-        s"[user:${activityUserName}] deleted branch ${branchName} at [repo:${userName}/${repositoryName}]",
-        None,
-        currentDate,
-        UUID.randomUUID().toString
-      )
-    )
-  }
-
-  def recordForkActivity(
-    userName: String,
-    repositoryName: String,
-    activityUserName: String,
-    forkedUserName: String
-  ): Unit = {
-    writeLog(
-      Activity(
-        userName,
-        repositoryName,
-        activityUserName,
-        "fork",
-        s"[user:${activityUserName}] forked [repo:${userName}/${repositoryName}] to [repo:${forkedUserName}/${repositoryName}]",
-        None,
-        currentDate,
-        UUID.randomUUID().toString
-      )
-    )
-  }
-
-  def recordPullRequestActivity(
-    userName: String,
-    repositoryName: String,
-    activityUserName: String,
-    issueId: Int,
-    title: String
-  ): Unit = {
-    writeLog(
-      Activity(
-        userName,
-        repositoryName,
-        activityUserName,
-        "open_pullreq",
-        s"[user:${activityUserName}] opened pull request [pullreq:${userName}/${repositoryName}#${issueId}]",
-        Some(title),
-        currentDate,
-        UUID.randomUUID().toString
-      )
-    )
-  }
-
-  def recordMergeActivity(
-    userName: String,
-    repositoryName: String,
-    activityUserName: String,
-    issueId: Int,
-    message: String
-  ): Unit = {
-    writeLog(
-      Activity(
-        userName,
-        repositoryName,
-        activityUserName,
-        "merge_pullreq",
-        s"[user:${activityUserName}] merged pull request [pullreq:${userName}/${repositoryName}#${issueId}]",
-        Some(message),
-        currentDate,
-        UUID.randomUUID().toString
-      )
-    )
-  }
-
-  def recordReleaseActivity(
-    userName: String,
-    repositoryName: String,
-    activityUserName: String,
-    releaseName: String,
-    tagName: String
-  ): Unit = {
-    writeLog(
-      Activity(
-        userName,
-        repositoryName,
-        activityUserName,
-        "release",
-        s"[user:${activityUserName}] released [release:${userName}/${repositoryName}/${tagName}:${releaseName}] at [repo:${userName}/${repositoryName}]",
-        None,
-        currentDate,
-        UUID.randomUUID().toString
-      )
-    )
-  }
-
-  private def cut(value: String, length: Int): String =
-    if (value.length > length) value.substring(0, length) + "..." else value
+  def recordActivity[T <: BaseActivityInfo](info: T): Unit =
```

Comment:
This is honestly the main benefit of this work. We no longer need to create a new function each time we want to log something. This function takes care of them all!